pkgs/development/libraries/hyperscan/default.nix

```diff
@@ -0,0 +1,71 @@
+{ lib, stdenv, fetchFromGitHub, cmake, ragel, python27
+, boost
+}:
+
+assert stdenv.isx86_64;  
+
+# NOTICE: pkgconfig, pcap and pcre intentionally omitted from build inputs
+#         pcap used only in examples, pkgconfig used only to check for pcre
+#         which is fixed 8.41 version requirement (nixpkgs have 8.42+, and
+#         I not see any reason (for now) to backport 8.41.
+
+stdenv.mkDerivation rec {
+  name = "${pname}-${version}";
+  pname = "hyperscan";
+  version = "5.0.0";
+
+  src = fetchFromGitHub {
+    owner = "intel";
+    repo = "hyperscan";
+    sha256 = "017dxg0n3gn9i4j27rcvpnp4rkqgycqni6x5d15dqpidl7zg7059";
+    rev = "v${version}";
+  };
+
+  outputs = [ "out" "dev" ];
+
+  buildInputs = [ boost ];
+  nativeBuildInputs = [ cmake ragel python27 ];
+
+  cmakeFlags = [
+    "-DFAT_RUNTIME=ON"
+    "-DBUILD_AVX512=ON"
+    "-DBUILD_STATIC_AND_SHARED=ON"
```

Comment:
Commonly that is exposed via a combination of `enableShared` and/or `enableStatic` options.

Reply:
I understand that -DBUILD_SHARED_LIBS=ON and -DBUILD_STATIC_LIBS=ON side by side is more readable, but it don't produce static libraries in this case. Their CMakeFIle looks pretty non-standard.